center_loss/face_model.py

```diff
@@ -0,0 +1,61 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.multiprocessing as mp
+import torch.utils.model_zoo as model_zoo
+from torchvision.models.resnet import BasicBlock, ResNet, model_urls
+import math
+
+class FaceModel(nn.Module):
+    def __init__(self,num_classes, pretrained=False, **kwargs):
+        super(FaceModel, self).__init__()
+        self.model = ResNet(BasicBlock, [2, 2, 2, 2], **kwargs)
+        if pretrained:
+            parameters =  model_zoo.load_url(model_urls['resnet50'])
+            self.model.load_state_dict(parameters)
+        self.model.avgpool = None
+        self.model.fc1 = nn.Linear(512*3*4, 512)
+        self.model.fc2 = nn.Linear(512, 512)
+        self.model.classifier = nn.Linear(512, num_classes)
+        self.centers = torch.zeros(num_classes, 512).type(torch.FloatTensor)
```

Comment:
You could define it as a `buffer` via `register_buffer`. This way, when you call `.cuda()`, `centers` is automatically sent to the GPU. Also, it's automatically loaded when you call `load_state_dict()`, and is also present in `state_dict()`.